core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1556,12 +1728,141 @@ public void warmUp() {
+            for (int i = columnCount, limit = detachedMetadata.getColumnCount(); i < limit; i++) {
+                int detColIdx = detachedMetadata.getWriterIndex(i);
+                String columnName = detachedMetadata.getColumnName(detColIdx);
+                detachedDeleteExtraColNames.add(columnName);
+            }
+
+            // check structure version
+            if (colsNegativeTypeMatch) {
+                if (metadata.getStructureVersion() == detachedMetadata.getStructureVersion()) {
+                    throw CairoException.detachedMetadataMismatch("structure_version should be different");
+                }
+            } else if (metadata.getStructureVersion() != detachedMetadata.getStructureVersion()) {
+                throw CairoException.detachedMetadataMismatch("structure_version");
+            }
+
+            // load/check _cv, updating local column tops
+            // set current _cv to where the partition was
+            other2.concat(COLUMN_VERSION_FILE_NAME).$();
+            if (!ff.exists(other2)) {
+                // Backups and older versions of detached partitions will not have _cv
+                LOG.info().$("detached _cv not found, skipping check [path=").$(other2).I$();
+                return;
+            }
+
+            if (detachedColumnVersionReader == null) {
+                detachedColumnVersionReader = new ColumnVersionReader();
+            }
+            detachedColumnVersionReader.ofRO(ff, other2);
+            detachedColumnVersionReader.readSafe(MillisecondClockImpl.INSTANCE, Long.MAX_VALUE);
+            LongList entries = columnVersionWriter.getCachedList();
+            LongList detachedEntries = detachedColumnVersionReader.getCachedList();
+            for (int i = 0, limit = detachedEntries.size(); i < limit; i += 4) {
+                // detachedEntries: [partitionTimestamp, columnIndex, columnNameTxn, columnTop]*
+                long columnTop = detachedEntries.getQuick(i + 3);
+                if (timestamp == detachedEntries.getQuick(i) && columnTop != entries.getQuick(i + 3)) {
+                    columnVersionWriter.upsert(
+                            timestamp,
+                            (int) detachedEntries.getQuick(i + 1),
+                            detachedEntries.getQuick(i + 2),
+                            columnTop
+                    );
+                }
+            }
+
+            // delete extra columns from .attachable (the table does not track/need them)
+            for (int i = 0, extraColNames = detachedDeleteExtraColNames.size(); i < extraColNames; i++) {
+                CharSequence columnName = detachedDeleteExtraColNames.get(i);
+                int colIdx = detachedMetadata.getColumnIndex(columnName);
+                long columnNameTxn = detachedColumnVersionReader.getColumnNameTxn(timestamp, colIdx);
+                removeFileAndOrLog(ff, dFile(other2.parent(), columnName, columnNameTxn));
```

Comment:
If you fail here, `columnVersionWriter` will have dirty entries which will be saved with the next commit.You better to write to column version files after all the file modifications

Reply:
I believe this to be the point where I remove extra files that the attaching partition had, if I don't fail at that it is no problem, just garbage in the fs, the columns will not exist in the table anyway